packages/signer/context-module/src/shared/model/ClearSignContext.ts

```diff
@@ -12,31 +12,66 @@ export enum ClearSignContextType {
   ERROR = "error",
 }
 
-export type ClearSignContextReference =
-  | {
-      type: ClearSignContextType.TOKEN | ClearSignContextType.NFT;
+export type ClearSignContextReferenceType =
+  | ClearSignContextType.TOKEN
+  | ClearSignContextType.NFT
+  | ClearSignContextType.ENUM
+  | ClearSignContextType.TRUSTED_NAME;
+
+export type ClearSignContextReference<
+  Type extends ClearSignContextReferenceType = ClearSignContextReferenceType,
+> = Type extends ClearSignContextType.ENUM
+  ? {
+      type: ClearSignContextType.ENUM;
       valuePath: GenericPath;
+      id: number; // enum id to reference
     }
-  | {
-      type: ClearSignContextType.TRUSTED_NAME;
-      valuePath: GenericPath;
-      types: string[];
-      sources: string[];
-    };
+  : Type extends ClearSignContextType.TRUSTED_NAME
+    ? {
+        type: ClearSignContextType.TRUSTED_NAME;
+        valuePath: GenericPath;
+        types: string[];
+        sources: string[];
+      }
+    : {
+        type: ClearSignContextType.TOKEN | ClearSignContextType.NFT;
```

Comment:
[MAYBE] Use parameter `Type` to get a concrete `ClearSignContextType`
```suggestion
        type: Type;
```
This will give out 
```ts
{ 
  type: ClearSignContextType.TOKEN;
  valuePath:  GenericPath;
}
```
if you pass `ClearSignContextType.TOKEN` as generic parameter which is more clear, I think. 
But this is not perfect including those two previous `extends` keywords, for example, the type 
```ts
ClearSignContextReference<ClearSignContextType.ENUM | ClearSignContextType.TRUSTED_NAME>
``` 
will give out 
```ts
{
  type: ClearSignContextType.ENUM;
  valuePath: GenericPath;
  id: number;
}
```
where you wanna get is the union of the two. Perhaps this does exist in use cases, but the typing here is not perfect narrowing the types.

Reply:
You are right, will add it 👍